test/unit/model/PermissionsServiceTest.php

```diff
@@ -251,6 +254,56 @@ public function testDoNotSaveAnythingWhenThereIsNothingToSave(): void
         $this->service->savePermissions(false, $resource, []);
     }
 
+    public function testDuplicatedAddPermissionsAreNotPersistedTwice(): void
+    {
+        $resource = $this->createMock(core_kernel_classes_Class::class);
+        $resource->method('getUri')->willReturn('res1');
+
+        // Mock a strategy returning a duplicated permission
+        //
```

Comment:
I think the test name (which it is a really good name) already explains that: 

`testDuplicatedAddPermissionsAreNotPersistedTwice`

Reply:
Removed 526f034f8899be0a63e1ae4cedeba7af3a202c4a